### ISSUE TITLE
Nightly test failure, gcc 14

### ISSUE THREAD

**Issue by Rombur** [kokkos/kokkos#7482]
Our nightly jenkins  fails see [here](https://cloud.cees.ornl.gov/jenkins-ci/job/Kokkos_nightly/302/pipeline-console/log?nodeId=53). The error is from `TestSIMD.cpp`. Maybe related to https://github.com/kokkos/kokkos/issues/7481

@ldh4 

```
[ 95%] Built target Kokkos_launch_bounds_reduce
In file included from /var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/TestSIMD.cpp:17:
In function 'void host_check_math_op_one_loader(BinaryOp, std::size_t, const T*, const T*) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; Loader = load_masked; BinaryOp = plus; T = int]',
    inlined from 'void host_check_math_op_all_loaders(Op, std::size_t, const T* ...) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; Op = plus; T = {int, int}]' at /var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/include/TestSIMD_MathOps.hpp:88:50,
    inlined from 'void host_check_all_math_ops(const DataType (&)[n], const DataType (&)[n]) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = int; long unsigned int n = 16]' at /var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/include/TestSIMD_MathOps.hpp:96:38:
/var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/include/TestSIMD_MathOps.hpp:48:31: error: 'first_arg' is used uninitialized [-Werror=uninitialized]
   48 |             binary_op.on_host(T(first_arg[lane]), T(second_arg[lane]));
      |                               ^~~~~~~~~~~~~~~~~~
/var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/include/TestSIMD_MathOps.hpp: In function 'void host_check_all_math_ops(const DataType (&)[n], const DataType (&)[n]) [with Abi = Kokkos::Experimental::simd_abi::avx2_fixed_size<4>; DataType = int; long unsigned int n = 16]':
/var/jenkins/workspace/Kokkos_nightly/simd/unit_tests/include/TestSIMD_MathOps.hpp:35:15: note: 'first_arg' declared here
   35 |     simd_type first_arg;
      |               ^~~~~~~~~
```

**Comment by ldh4** [comment]
#7479 is the PR that aims to resolve this issue.

**Comment by ldh4** [comment]
This should be resolved by #7484 for now.
